worker/src/main/java/gov/cms/ab2d/worker/processor/JobProcessorImpl.java

```diff
@@ -127,26 +127,82 @@ void processContract(Job job, Path outputDirPath)
         Contract contract = job.getContract();
         assert contract != null;
         log.info("Job [{}] - contract [{}] ", job.getJobUuid(), contract.getContractNumber());
-        // Retrieve the contract beneficiaries
 
-        Map<Long, CoverageSummary> patients = processContractBenes(job);
+        try {
+            // Retrieve the contract beneficiaries
+            Map<Long, CoverageSummary> patients = processContractBenes(job);
+
+            // Create a holder for the contract, writer, progress tracker and attested date
+            JobData jobData = new JobData(job.getJobUuid(), job.getSince(), getOrganization(job), patients);
 
-        // Create a holder for the contract, writer, progress tracker and attested date
-        JobData jobData = new JobData(job.getJobUuid(), job.getSince(), getOrganization(job), patients);
+            var jobOutputs = contractProcessor.process(outputDirPath, job, jobData);
 
-        var jobOutputs = contractProcessor.process(outputDirPath, job, jobData);
+            // For each job output, add to the job and save the result
+            jobOutputs.forEach(job::addJobOutput);
+            jobOutputRepository.saveAll(jobOutputs);
 
-        // For each job output, add to the job and save the result
-        jobOutputs.forEach(job::addJobOutput);
-        jobOutputRepository.saveAll(jobOutputs);
+            // If the job is done searching
+            verifyTrackedJobProgress(job, contract);
+        } finally {
+            // Guarantee that we write out statistics on the job if possible
+            persistTrackedJobProgress(job, contract);
```

Comment:
Will write out even if job fails or is cancelled to give us better metrics when jobs fail.